File: algorithms/2_medium/M/03239_minimum_number_of_flips_to_make_binary_grid_palindromic_i/solution.py
class Solution:
    def minFlips(self, grid: List[List[int]]) -> int:
        rows = len(grid)
        cols = len(grid[0])
        row_cost = 0
        col_cost = 0

        for i in range(rows // 2):
            other_i = rows - 1 - i

            for j in range(cols):
                if grid[i][j] != grid[other_i][j]:
                    row_cost += 1

        for j in range(cols // 2):
            other_j = cols - 1 - j

            for i in range(rows):
                if grid[i][j] != grid[i][other_j]:
                    col_cost += 1

        return min(col_cost, row_cost)

File: algorithms/2_medium/M/03239_minimum_number_of_flips_to_make_binary_grid_palindromic_i/solution.cpp
class Solution {
public:
    int minFlips(vector<vector<int>>& grid) {
        const size_t rows{ grid.size() };
        const size_t cols{ grid[0].size() };
        int row_cost{ 0 };
        int col_cost{ 0 };

        for (size_t i{ 0 }; i < rows / 2; ++i) {
            const size_t other_i{ rows - 1 - i };

            for (size_t j{ 0 }; j < cols; ++j) {
                if (grid[i][j] != grid[other_i][j]) {
                    row_cost += 1;
                }
            }
        }

        for (size_t j{ 0 }; j < cols / 2; ++j) {
            const size_t other_j{ cols - 1 - j };

            for (size_t i { 0 }; i < rows; ++i) {
                if (grid[i][j] != grid[i][other_j]) {
                    col_cost += 1;
                }
            }
        }

        return std::min(col_cost, row_cost);
    }
};
